HackAtUCI/irvinehacks-site
# Redesign data model for volunteer availability

As part of #477, simplifying the data processing for volunteer availability to remove unnecessary manual validation and make it easier to use in the future.

Previous data format
```json
	"friday_availability": {
		"11PM": false,
		"10PM": false,
		"9PM": false,
		"8PM": false,
		"7PM": false,
		"6PM": false,
		"5PM": true,
		"4PM": false,
		"3PM": false,
		"2PM": false,
		"1PM": true,
		"12PM": false,
		"11AM": true,
		"10AM": false,
		"9AM": false,
		"8AM": false,
		"7AM": false
	},

```

New data format
```json
	"friday_availability": [11, 13, 17],
```

### Changes
- Simplify data handling for volunteer time slots with name-shared input
  - Works like multi-select checkboxes and validated as a list of values
  - Provide list of numeric hours instead of mapping from 12-hour label
- Revert complications to apply flow while adding extraneous validation in #474
- Fix typo with `$exists` selector for checking already applied (described in #466)
- Extract global variable for fields supporting "other" values

### Testing
1. Run all development servers including the local database
2. Temporarily disable emails or use the dev mock being added in #465
3. For each type of applicant (Hacker, Mentor, Volunteer), impersonate a new user and complete an application
4. Observe the data is generally processed as expected (besides known bugs)
5. Check the database for the new entries and confirm the volunteer availability looks as expected
